core/src/main/java/org/opensearch/sql/utils/JsonUtils.java

```diff
@@ -23,9 +47,101 @@ public static ExprValue isValidJson(ExprValue jsonExprValue) {
 
     try {
       objectMapper.readTree(jsonExprValue.stringValue());
-      return ExprValueUtils.LITERAL_TRUE;
+      return LITERAL_TRUE;
     } catch (JsonProcessingException e) {
-      return ExprValueUtils.LITERAL_FALSE;
+      return LITERAL_FALSE;
+    }
+  }
+
+  /**
+   * Converts a JSON encoded string to a {@link ExprValue}. Expression type will be UNDEFINED.
+   *
+   * @param json JSON string (e.g. "{\"hello\": \"world\"}").
+   * @return ExprValue returns an expression that best represents the provided JSON-encoded string.
+   *     <ol>
+   *       <li>{@link ExprTupleValue} if the JSON is an object
+   *       <li>{@link ExprCollectionValue} if the JSON is an array
+   *       <li>{@link ExprDoubleValue} if the JSON is a floating-point number scalar
+   *       <li>{@link ExprIntegerValue} if the JSON is an integral number scalar
+   *       <li>{@link ExprStringValue} if the JSON is a string scalar
+   *       <li>{@link ExprBooleanValue} if the JSON is a boolean scalar
+   *       <li>{@link ExprNullValue} if the JSON is null, empty, or invalid
+   *     </ol>
+   */
+  public static ExprValue castJson(ExprValue json) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode jsonNode;
+    try {
+      jsonNode = objectMapper.readTree(json.stringValue());
+    } catch (JsonProcessingException e) {
+      final String errorFormat = "JSON string '%s' is not valid. Error details: %s";
+      throw new SemanticCheckException(String.format(errorFormat, json, e.getMessage()), e);
+    }
+
+    return processJsonNode(jsonNode);
+  }
+
+  /**
+   * Extract value of JSON string at given JSON path.
+   *
+   * @param json JSON string (e.g. "{\"hello\": \"world\"}").
+   * @param path JSON path (e.g. "$.hello")
+   * @return ExprValue of value at given path of json string.
+   */
+  public static ExprValue extractJson(ExprValue json, ExprValue path) {
+    if (json == LITERAL_NULL || json == LITERAL_MISSING) {
+      return json;
+    }
+
+    String jsonString = json.stringValue();
+    String jsonPath = path.stringValue();
+
+    if (jsonString.isEmpty()) {
+      return LITERAL_NULL;
+    }
+
+    try {
+      Object results = JsonPath.parse(jsonString).read(jsonPath);
+      return ExprValueUtils.fromObjectValue(results);
+    } catch (PathNotFoundException e) {
+      return LITERAL_NULL;
+    } catch (InvalidPathException e) {
```

Comment:
please don't reuse variable names:
```suggestion
    } catch (InvalidPathException invalidPathException) {
```

Reply:
addressed